contributed_definitions/nyaml/NXopt_element.yaml

```diff
@@ -0,0 +1,36 @@
+category: base 
+doc: |
+  Properties of generic optical element in an experimental setup for optical spectroscopy.
+
+  
+NXopt_element(NXobject):
+
+  previous_optical_elements: 

```

Comment:
If you don't write any type here, the type will default to NX_CHAR (essentially a string). This doesn't work too well if you want to allow lists of elements.

Reply:
Actually, every field in NeXus is a scalar or an array if not specified differently.